doc_src/prompt.rst

```diff
@@ -0,0 +1,189 @@
+
+  function fish_prompt
+      string join '' -- (set_color green) (prompt_pwd -D 2) (set_color normal) '>'
+  end
+
+With a current directory of "/home/tutorial/Music/Lena Raine/Oneknowing", this would print
+
+.. role:: green
+.. parsed-literal::
+    :class: highlight
+
+    :green:`~/M/Lena Raine/Oneknowing`>_
+
+Status
+------
+
+One important bit [#]_ of information that every command returns is the :ref:`status <variables-status>`. This is either 0 if the command returned successfully, or a whole number from 1 to 255 if not, so it's best understood as an error code.
+
+It's useful to display this in your prompt, but showing it when it's 0 seems kind of wasteful.
+
+First of all, since every command (except for :doc:`set <cmds/set>`) changes the status, you need to store it for later use as the first thing in your prompt. Use a local variable so it will be confined to your prompt function::
+
+  set -l last_status $status
+  
+And after that, you can set a string if it not zero::
+  
+  # Prompt status only if it's not 0
+  set -l stat
+  if test $last_status -ne 0
+      set stat (set_color red)"[$last_status]"(set_color normal)
+  end
+
+And to print it, we add it to our ``string join``::
+
+  string join '' -- (set_color green) (prompt_pwd) (set_color normal) $stat '>'
+  
+In case that $last_status was 0, $stat is empty, and so it will simply disappear.
+
+So our entire prompt is now::
+
+  function fish_prompt
+      set -l last_status $status
+      # Prompt status only if it's not 0
+      set -l stat
+      if test $last_status -ne 0
+          set stat (set_color red)"[$last_status]"(set_color normal)
+      end
+
+      string join '' -- (set_color green) (prompt_pwd) (set_color normal) $stat '>'
+  end
+
+And it looks like:
+
+.. role:: green
+.. parsed-literal::
+    :class: highlight
+
+    :green:`~/M/L/Oneknowing`\ :red:`[1]`>_
+
+after we run ``false`` (which returns 1).
+
+.. [#] Actually a byte, i.e. 8 bits - it can go from 0 to 255
+
+Where to go from here?
+----------------------
+
+We have now built a simple but working and usable prompt, but of course more can be done.
+
+- Fish offers more helper functions:
+  - ``prompt_login`` to describe the user/hostname/container or ``prompt_hostname`` to describe just the host
+  - ``fish_is_root_user`` to help with changing the symbol for root.
+  - ``fish_vcs_prompt`` to show version control information (or ``fish_git_prompt`` / ``fish_hg_prompt`` / ``fish_svn_prompt`` to limit it to specific systems)
+- You can add a right prompt by changing :doc:`fish_right_prompt <cmds/fish_right_prompt>` or a vi-mode prompt by changing :doc:`fish_mode_prompt <cmds/fish_mode_prompt>`.
+- Some prompts have interesting or advanced features
+  - Add a date?
+  - Show various integrations like python's venv
+  - Color the parts differently.
+
+You can look at fish's sample prompts for inspiration.
```

Comment:
Is there any way of linking to these? What about mentioning `fish_config` here?

Reply:
Done, good idea!